op-node/rollup/derive/span_channel_out.go

```diff
@@ -130,6 +133,12 @@ func (co *SpanChannelOut) AddSingularBatch(batch *SingularBatch, seqNum uint64)
 	if err := co.spanBatch.AppendSingularBatch(batch, seqNum); err != nil {
 		return fmt.Errorf("failed to append SingularBatch to SpanBatch: %w", err)
 	}
+	defer func() {
+		if err != nil {
+			co.spanBatch.revertLastBatch()
+		}
+	}()
```

Comment:
Span Channel Out already blocks new Batches from being added if it is already full, so I'm not sure why we'd want to revert the spanBatch. I guess if there was a failure to `rlp` encode or something, this would roll back that batch, but this revert would quietly drop a batch from a channel, which the Channel Builder now needs to understand and handle.

I could see a situation where reverting makes sense -- you wouldn't want to add the same batch to the SpanBatch twice, and then have that be part of the encoded structure. But I could also see this complexity lead to batches being dropped -- if the error being returned is `FullErr`, this revert would mutate the underlying structure (which could be okay since it is already encoded and compressed, but still has a smell to it)

Reply:
In ChannelBuilder, `block` is only [added](https://github.com/ethereum-optimism/optimism/blob/53afefdd2e53dac3e7597e58bb78132ccee9d2c4/op-batcher/batcher/channel_builder.go#L166) if `AddSingularBatch` doesn't return an error, so I think it's better to be consistent that if an error is returned, `co.spanBatch` should revert to its previous state.